compiler/qsc/src/interpret/stateful.rs

```diff
@@ -361,86 +313,91 @@ impl Interpreter {
     pub fn interpret_line(&mut self, receiver: &mut impl Receiver, line: &str) -> LineResult {
         let mut result = Value::unit();
 
-        let label = self.next_line_label();
-        let mut fragments = self
-            .compiler
-            .compile_fragments(&label, line)
-            .map_err(|errors| {
-                errors
-                    .into_iter()
-                    .map(|error| {
-                        LineError(WithSource::from_map(
-                            self.compiler.source_map(),
-                            error.into(),
-                        ))
-                    })
-                    .collect::<Vec<_>>()
-            })?;
+        let label = &self.next_line_label();
+
+        let (items, stmts) =
+            self.compile_incremental(|c, p| c.compile_fragments(p, label, line))?;
```

Comment:
I extracted a helper function here `compile_incremental` for code repeated in three functions (`interpret_line`, `run`, `qirgen`). In the future I'm hoping that bit of code will become part of the incremental compiler I separate out from this file.

Reply:
Agreed; I think it could make sense to pull that up into either a new `incremental.rs` or the existing `compile.rs` that sits one folder up in qsc/src